File: client/java/src/main/java/com/be/mesh/client/struct/Route.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.struct;

import com.be.mesh.client.annotate.Index;
import lombok.Data;

import java.io.Serializable;

/**
 * @author coyzeng@gmail.com
 */
@Data
public class Route implements Serializable {

    private static final long serialVersionUID = 9210017214247532109L;
    /**
     * 节点编号
     */
    @Index(value = 0, name = "node_id")
    private String nodeId;
    /**
     * 机构编号
     */
    @Index(value = 5, name = "inst_id")
    private String instId;
    /**
     * 节点名称
     */
    @Index(10)
    private String name;
    /**
     * 机构名称
     */
    @Index(value = 15, name = "inst_name")
    private String instName;
    /**
     * 节点地址
     */
    @Index(20)
    private String address;
    /**
     * 节点描述
     */
    @Index(25)
    private String describe;
    /**
     * Host root certifications.
     */
    @Index(value = 30, name = "host_root")
    private String hostRoot;
    /**
     * Host private certifications key
     */
    @Index(value = 35, name = "host_key")
    private String hostKey;
    /**
     * Host certification
     */
    @Index(value = 40, name = "host_crt")
    private String hostCrt;
    /**
     * Guest root certifications
     */
    @Index(value = 45, name = "guest_root")
    private String guestRoot;
    /**
     * Guest private certifications key.
     */
    @Index(value = 50, name = "guest_key")
    private String guestKey;
    /**
     * Guest certifications key.
     */
    @Index(value = 55, name = "guest_crt")
    private String guestCrt;
    /**
     * 状态
     */
    @Index(60)
    private int status;
    /**
     * 版本
     */
    @Index(65)
    private int version;
    /**
     * Auth code
     */
    @Index(value = 70, name = "auth_code")
    private String authCode;
    /**
     * Expire time
     */
    @Index(value = 75, name = "expire_at")
    private Long expireAt;
    /**
     * Extra info
     */
    @Index(80)
    private String extra;
    /**
     * Node status
     */
    @Index(value = 85, name = "create_at")
    private String createAt;
    /**
     * Node status
     */
    @Index(value = 90, name = "create_by")
    private String createBy;
    /**
     * Node status
     */
    @Index(value = 95, name = "update_at")
    private String updateAt;
    /**
     * Node status
     */
    @Index(value = 100, name = "update_by")
    private String updateBy;
    /**
     * Network group.
     */
    @Index(value = 105)
    private String group;
    /**
     * Upstream bandwidth.
     */
    @Index(value = 110)
    private long upstream;
    /**
     * Downstream bandwidth.
     */
    @Index(value = 115)
    private long downstream;
    /**
     * Static public ip address.
     */
    @Index(value = 120, name = "static_ip")
    private String staticIP;
    /**
     * Proxy endpoint in transport
     */
    @Index(125)
    private String proxy;
    /**
     * MPC concurrency
     */
    @Index(130)
    private long concurrency;
}

File: client/python/mesh/kinds/route.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from mesh.macro import index, serializable


@serializable
class Route:

    @index(0)
    def node_id(self) -> str:
        """ 节点编号 """
        return ""

    @index(5)
    def inst_id(self) -> str:
        """ 机构编号 """
        return ""

    @index(10)
    def name(self) -> str:
        """ 节点名称 """
        return ""

    @index(15)
    def inst_name(self) -> str:
        """ 机构编号 """
        return ""

    @index(20)
    def address(self) -> str:
        """ 节点地址 """
        return ""

    @index(25)
    def describe(self) -> str:
        """ 节点描述 """
        return ""

    @index(30)
    def host_root(self) -> str:
        """ Host root certifications """
        return ""

    @index(35)
    def host_key(self) -> str:
        """ Host private certifications key """
        return ""

    @index(40)
    def host_crt(self) -> str:
        """ Host certification """
        return ""

    @index(45)
    def guest_root(self) -> str:
        """ Guest root certifications """
        return ""

    @index(50)
    def guest_key(self) -> str:
        """ Guest private certifications key """
        return ""

    @index(55)
    def guest_crt(self) -> str:
        """ Guest certifications key """
        return ""

    @index(60)
    def status(self) -> int:
        """ 状态 """
        return 0

    @index(65)
    def version(self) -> int:
        """ 状态 """
        return 0

    @index(70)
    def auth_code(self) -> str:
        """ 授权码 """
        return ""

    @index(75)
    def expire_at(self) -> int:
        """ 失效时间 """
        return 0

    @index(80)
    def extra(self) -> str:
        """ 扩展信息 """
        return ""

    @index(85)
    def create_at(self) -> str:
        """ 创建时间 """
        return ""

    @index(90)
    def create_by(self) -> str:
        """ 创建人 """
        return ""

    @index(95)
    def update_at(self) -> str:
        """ 更新时间 """
        return ""

    @index(100)
    def update_by(self) -> str:
        """ 跟新人 """
        return ""

    @index(105)
    def group(self) -> str:
        """ Network group. """
        return ""

    @index(110)
    def upstream(self) -> int:
        """ Upstream bandwidth. """
        return 0

    @index(115)
    def downstream(self) -> int:
        """ Downstream bandwidth. """
        return 0

    @index(120)
    def static_ip(self) -> str:
        """ Static public ip address """
        return ""

    @index(125)
    def proxy(self) -> str:
        """ Proxy endpoint in transport """
        return ""

    @index(130)
    def concurrency(self) -> int:
        """ MPC concurrency """
        return 0
